### PR TITLE
feat: initial version of the library

### PR DESCRIPTION
**Pull request by smira**
Most of the code is coming directly from Talos `pkg/kubernetes` and `pkg/cluster/kubernetes`, but with some changes to make it more "standalone" so that it has less dependencies.

### CODE REVIEW

**Review by frezbo — APPROVED** [review]

File: kubernetes/upgrade/checks.go [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+	}
+
+	err = json.Unmarshal(jsonSerialized, &spec)
+
+	return &spec, err
+}
+
+// Error returns the error message.
+func (e ComponentRemovedItemsError) Error() string {
+	var buf bytes.Buffer
```
Comment (DmitriyMV) [inline kubernetes/upgrade/checks.go:296]:
```suggestion
	var buf strings.Builder
```

Comment (smira) [reply kubernetes/upgrade/checks.go]: thanks, updated

File: kubernetes/manifests/sync.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+	ignoreKey("metadata", "resourceVersion")
+	ignoreKey("metadata", "generation")
+	ignoreKey("metadata", "creationTimestamp")
+	ignoreKey("metadata", "deletionTimestamp")
+	ignoreKey("metadata", "deletionGracePeriodSeconds")
+	ignoreKey("metadata", "managedFields")
+	ignoreKey("metadata", "finalizers")
+	ignoreKey("metadata", "selfLink")
+	ignoreKey("metadata", "ownerReferences")
+	ignoreKey("metadata", "annotations")
```
Comment (frezbo) [inline kubernetes/manifests/sync.go:222]: this is a user provided field, right?

**Review by frezbo — APPROVED** [review]

File: kubernetes/manifests/sync.go (hunk 2 above)
Comment (smira) [reply kubernetes/manifests/sync.go:222]: yep, but Kubernetes puts stuff there for Deployments... and it makes us remove that annotation. probably should go deeper here, like ignore only some annotations?

Comment (smira) [reply kubernetes/manifests/sync.go]: working on a more specific version here....

Comment (smira) [reply kubernetes/manifests/sync.go]: now it should be better, it ignores Kubernetes annotations, but allows to update user annotations.

**Pull request merged**